converter/convert.go

```diff
@@ -79,23 +80,59 @@ func ConvertBundleToOCIIndex(b *bundle.Bundle, targetRef reference.Named, bundle
 	return &result, nil
 }
 
-// ConvertOCIIndexToBundle converts an OCI index to a CNAB bundle representation
-func ConvertOCIIndexToBundle(ix *ocischemav1.Index, config *BundleConfig, originRepo reference.Named) (*bundle.Bundle, error) {
-	b := &bundle.Bundle{
-		SchemaVersion: CNABVersion,
-		Actions:       config.Actions,
-		Credentials:   config.Credentials,
-		Definitions:   config.Definitions,
-		Parameters:    config.Parameters,
-		Custom:        config.Custom,
-	}
-	if err := parseTopLevelAnnotations(ix.Annotations, b); err != nil {
-		return nil, err
-	}
-	if err := parseManifests(ix.Manifests, b, originRepo); err != nil {
-		return nil, err
+// GenerateRelocationMap TODO
+func GenerateRelocationMap(ix *ocischemav1.Index, b *bundle.Bundle, originRepo reference.Named) (bundle.ImageRelocationMap, error) {
+	relocationMap := bundle.ImageRelocationMap{}
+
+	for _, d := range ix.Manifests {
+		switch d.MediaType {
+		case ocischemav1.MediaTypeImageManifest, ocischemav1.MediaTypeImageIndex:
+		case images.MediaTypeDockerSchema2Manifest, images.MediaTypeDockerSchema2ManifestList:
+		default:
+			return nil, fmt.Errorf("unsupported manifest descriptor %q with mediatype %q", d.Digest, d.MediaType)
+		}
+		descriptorType, ok := d.Annotations[CNABDescriptorTypeAnnotation]
+		if !ok {
+			return nil, fmt.Errorf("manifest descriptor %q has no CNAB descriptor type annotation %q", d.Digest, CNABDescriptorTypeAnnotation)
+		}
+		if descriptorType == CNABDescriptorTypeConfig {
+			continue
+		}
+		// strip tag/digest from originRepo
+		originRepo, err := reference.ParseNormalizedNamed(originRepo.Name())
+		if err != nil {
+			return nil, fmt.Errorf("failed to create a digested reference for manifest descriptor %q: %s", d.Digest, err)
+		}
+		ref, err := reference.WithDigest(originRepo, d.Digest)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create a digested reference for manifest descriptor %q: %s", d.Digest, err)
+		}
+		refFamiliar := reference.FamiliarString(ref)
+		switch descriptorType {
+		// The current descriptor is an invocation image
+		case CNABDescriptorTypeInvocation:
+			if len(b.InvocationImages) == 0 {
+				return nil, fmt.Errorf("unknown invocation image: %q", d.Digest)
+			}
+			relocationMap[b.InvocationImages[0].Image] = refFamiliar
```

Comment:
Here we still only handle the first invocation image from a bundle - at some future point we should handle multiple invocation images.

Reply:
Agreed. Is the current restriction captured in an issue?